File: frontend/src/types/report.ts
// 신고 대상의 타입을 정의하는 Enum
// 백엔드의 ReportType Enum과 반드시 일치해야 합니다.
export enum ReportType {
    POST = 'POST',
    COMMENT = 'COMMENT',
    STUDY_GROUP = 'STUDY_GROUP',
}

// (선택 사항) 신고 API 요청 시 사용될 DTO 타입
export interface ReportRequest {
    reportType: ReportType;
    targetId: number;
    reason: string;
}

// (선택 사항) 관리자 페이지에서 신고 목록을 받을 때 사용할 타입
export interface ReportResponse {
    id: number;
    reporter: {
        id: number;
        name: string;
    };
    reportType: ReportType;
    targetId: number;
    reportedUser?: {
        id: number;
        name: string;
    };
    reason: string;
    status: 'RECEIVED' | 'IN_PROGRESS' | 'COMPLETED';
    adminMemo?: string;
    createdAt: string;
}

// --- 이 Enum을 추가합니다 ---
export enum ReportStatus {
    RECEIVED = 'RECEIVED',
    IN_PROGRESS = 'IN_PROGRESS',
    COMPLETED = 'COMPLETED',
}

// ReportDetailDto 타입 정의 추가
export interface ReportDetailDto {
    id: number;
    reportType: ReportType;
    targetId: number;
    targetContentPreview: string;
    reason: string;
    status: ReportStatus;
    reporterName: string;
    reportedUserName: string;
    createdAt: string;
    adminMemo: string;
}

File: src/main/java/com/studygroup/domain/report/entity/Report.java
package com.studygroup.domain.report.entity;

import com.studygroup.domain.user.entity.User;
import com.studygroup.global.common.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Report extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 신고한 사용자
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reporter_id", nullable = false)
    private User reporter;

    // 신고 대상의 타입 (게시글, 댓글 등)
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportType reportType;

    // 신고 대상의 ID
    @Column(nullable = false)
    private Long targetId;

    // (선택) 신고 대상 콘텐츠의 작성자
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reported_user_id")
    private User reportedUser;

    @Lob
    @Column(nullable = false)
    private String reason; // 신고 사유

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReportStatus status;

    @Lob
    private String adminMemo; // 관리자 처리 메모

    @Builder
    public Report(User reporter, ReportType reportType, Long targetId, User reportedUser, String reason) {
        this.reporter = reporter;
        this.reportType = reportType;
        this.targetId = targetId;
        this.reportedUser = reportedUser;
        this.reason = reason;
        this.status = ReportStatus.RECEIVED; // 최초 상태는 '접수됨'
    }

    public void updateStatus(ReportStatus status, String adminMemo) {
        this.status = status;
        this.adminMemo = adminMemo;
    }
}
